src/clj/game/cards/programs.clj

```diff
@@ -435,10 +435,20 @@
    "Imp"
    {:flags {:slow-trash (req (pos? (get-in card [:counter :virus] 0)))}
     :data {:counter {:virus 2}}
-    :abilities [{:counter-cost [:virus 1]
-                 :msg "trash at no cost"
-                 :once :per-turn
-                 :effect (effect (trash-no-cost))}]}
+    :events {:access {:delayed-completion true
+                      :req (req (pos? (get-in card [:counter :virus] 0)))
+                      :once :per-turn
```

Comment:
I don't think this belongs here?  The once per turn relates to actual firing of the trash effect which is in the effect clause below.  This should be an additional :req check to ensure the trash has not been used yet.